de/nmichael/efa/data/storage/MetaData.java

```diff
@@ -130,8 +130,20 @@ public int getFieldType(int i) {
         return TYPES[i];
     }
 
+    /**
+     * Returns the iDataAccess field type of a given field.
+     * If fieldName does not exists, it returns iDataAccess.DATA_UNKOWN.
+     * 
+     * @param fieldName
+     * @return iDataAccess field type value (int). If fieldName does not exist, it returns IDataAccess.DATA_UNKOWN.
+     */
     public int getFieldType(String fieldName) {
-        return getFieldType(getFieldIndex(fieldName));
+        int fieldIndex = getFieldIndex(fieldName);
+    	if (fieldIndex>=0) {
+    		return getFieldType(fieldIndex);
+    	} else {
+    		return IDataAccess.DATA_UNKNOWN;
```

Comment:
It could be more elegant to leave getFieldType(String) as is and instead modify getFieldType(int) in line 130 to ```return` i >= 0 ? TYPES[i] : IDataAccess.DATA_UNKNOWN;```
But this would work too.